test/smoke_test/smoke_test.py

```diff
@@ -3,29 +3,78 @@
 import torch
 import torchvision
 import torchaudio
+import pathlib
 
-def smoke_test_cuda() -> None:
-    gpu_arch_ver = os.getenv('GPU_ARCH_VER')
-    gpu_arch_type = os.getenv('GPU_ARCH_TYPE')
-    is_cuda_system = gpu_arch_type == "cuda"
+gpu_arch_ver = os.getenv('GPU_ARCH_VER')
+gpu_arch_type = os.getenv('GPU_ARCH_TYPE')
+is_cuda_system = gpu_arch_type == 'cuda'
 
+def smoke_test_cuda() -> None:
     if(not torch.cuda.is_available() and is_cuda_system):
-        print(f"Expected CUDA {gpu_arch_ver}. However CUDA is not loaded.")
-        sys.exit(1)
+        raise RuntimeError(f'Expected CUDA {gpu_arch_ver}. However CUDA is not loaded.')
     if(torch.cuda.is_available()):
         if(torch.version.cuda != gpu_arch_ver):
-            print(f"Wrong CUDA version. Loaded: {torch.version.cuda} Expected: {gpu_arch_ver}")
-            sys.exit(1)
-        y=torch.randn([3,5]).cuda()
-        print(f"torch cuda: {torch.version.cuda}")
+            raise RuntimeError(f'Wrong CUDA version. Loaded: {torch.version.cuda} Expected: {gpu_arch_ver}')
+        print(f'torch cuda: {torch.version.cuda}')
         #todo add cudnn version validation
-        print(f"torch cudnn: {torch.backends.cudnn.version()}")
+        print(f'torch cudnn: {torch.backends.cudnn.version()}')
+
+def smoke_test_conv2d() -> None:
+    import torch.nn as nn
+    print('Calling smoke_test_conv2d')
+    # With square kernels and equal stride
+    m = nn.Conv2d(16, 33, 3, stride=2)
+    # non-square kernels and unequal stride and with padding
+    m = nn.Conv2d(16, 33, (3, 5), stride=(2, 1), padding=(4, 2))
+    # non-square kernels and unequal stride and with padding and dilation
+    m = nn.Conv2d(16, 33, (3, 5), stride=(2, 1), padding=(4, 2), dilation=(3, 1))
+    input = torch.randn(20, 16, 50, 100)
+    output = m(input)
+    if(is_cuda_system):
+        print('Testing smoke_test_conv2d with cuda')
+        conv = nn.Conv2d(3, 3, 3).cuda()
+        x = torch.randn(1, 3, 24, 24).cuda()
+        with torch.cuda.amp.autocast():
+            out = conv(x)
 
 def smoke_test_torchvision() -> None:
-    import torchvision.datasets as dset
-    import torchvision.transforms
     print('Is torchvision useable?', all(x is not None for x in [torch.ops.image.decode_png, torch.ops.torchvision.roi_align]))
 
+def smoke_test_torchvision_read_decode() -> None:
+    from torchvision.io import read_image
+    img_jpg = read_image(str(pathlib.Path(__file__).parent / 'assets' / 'rgb_pytorch.jpg'))
+    assert img_jpg.ndim == 3 and img_jpg.numel() > 100
```

Comment:
Nit: (Python runtime can be configured to ignore asserts, but test is supposed to always raise an exception)
```suggestion
    if img_jpg.ndim != 3 or img_jpg.numel() < 100:
      raise RuntimeError(f'Unexpected shape of img_rgb: {img_rgb.shape}')
```